### PR TITLE
feat(inventory): add dedicated right for locked field

### PR DESCRIPTION
**Pull request by stonebuzz**
Add dedicated right for locked field (only ```CREATE``` (for global lock) and ```PURGE``` are needed)

image

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #number

### CODE REVIEW

**Review by orthagh — APPROVED** [review]

**Comment by cedric-anne** [comment]
Can you check why the `PURGE` right is not placed in same column as `PURGE` right of other objects ?

**Comment by stonebuzz** [comment]
Fixed 
image

**Review by trasher — APPROVED** [review]

File: src/Lockedfield.php [hunk 1]
```diff
@@ -251,7 +251,7 @@ public function getForbiddenStandardMassiveAction()
 
     public static function canPurge()
     {
-        return Session::haveRight(static::$rightname, UPDATE);
+        return Session::haveRight(static::$rightname, PURGE);
     }
```
Comment (cedric-anne) [inline src/Lockedfield.php:255]: This method is probably useless now, as the default implementation can be used now.

File: src/Lockedfield.php [hunk 2]
```diff
@@ -291,7 +291,7 @@ public function showForm($ID, array $options = [])
     public static function canCreate()
     {
         if (static::$rightname) {
-            return Session::haveRight(static::$rightname, UPDATE);
+            return Session::haveRight(static::$rightname, CREATE);
         }
         return false;
     }
```
Comment (cedric-anne) [inline src/Lockedfield.php:297]: This method is probably useless now, as the default implementation can be used now.

**Review by cedric-anne — CHANGES_REQUESTED** [review]

File: src/Lockedfield.php (hunk 1 above)
Comment (stonebuzz) [reply src/Lockedfield.php]: indeed

File: src/Lockedfield.php (hunk 2 above)
Comment (stonebuzz) [reply src/Lockedfield.php]: indeed

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**